### PR TITLE
Docs (Support): Clarify support policy for releases, communicate API / service support policy

### PR DESCRIPTION
**Pull request by aaroncarey**
This change clarifies our support policy for releases and communicates how we plan to handle breaking API changes, underlying data format changes, and rolling upgrades that don't impact APIs or the underlying data format.

Open issues:
- How much advance notice will we provide for breaking API changes (need to spell out our minimum that we'll commit to).
- How will we provide advance notice to customers about breaking API changes. Using Discuss, email communication, etc.?
-"Upgrade window": What can we commit to, in terms of how long the upgrade window is?
- "Switch to read-only": How soon will we be able to switch Dgraph Cloud backends to "read-only" during upgrades? Currently they are 100% offline for at least a few seconds during upgrades.
- What happens in the case of an outage? How quickly can customers expect resolution?

Fixes DOC-269

### CODE REVIEW

File: content/releases/index.md [hunk 1]
```diff
@@ -39,3 +39,29 @@ To learn about the latest releases and other important announcements, watch the
...
+
+## Dgraph support policy
+
+The following summarizes our approach to product and service support:
+ 
+* **Dgraph releases**: Dgraph Labs supports Dgraph releases for 12 months following the
+ first release date, until the applicable *End of life* date (see above).
+* **Breaking API changes**: Occasionally, a new Dgraph release or Dgraph Cloud
+service update will include breaking API changes. When this happens, Dgraph Labs
+will provide advance notice to Dgraph customers so they can update their code
```
Comment (aaroncarey) [inline content/releases/index.md:51]: Two questions that we need to answer here: 
* How much notice, at a minimum, should we commit to?
* Will we announce the breaking change over Discuss, or will we also contact customers directly using email?

File: content/releases/index.md [hunk 2]
```diff
@@ -39,3 +39,29 @@ To learn about the latest releases and other important announcements, watch the
...
+ 
+* **Dgraph releases**: Dgraph Labs supports Dgraph releases for 12 months following the
+ first release date, until the applicable *End of life* date (see above).
+* **Breaking API changes**: Occasionally, a new Dgraph release or Dgraph Cloud
+service update will include breaking API changes. When this happens, Dgraph Labs
+will provide advance notice to Dgraph customers so they can update their code 
+to work with the breaking API change.
+* **Data format changes**: Occasionally, Dgraph Cloud service updates will include
+changes to the underlying data format. When this happens, Dgraph Labs will contact
+Dgraph Cloud customers to schedule an upgrade window. During the upgrade,
```
Comment (aaroncarey) [inline content/releases/index.md:55]: Can we, or do we want to, specify how long the upgrade window is?

File: content/releases/index.md [hunk 3]
```diff
@@ -39,3 +39,29 @@ To learn about the latest releases and other important announcements, watch the
...
+your Dgraph Cloud backend will switch to read-only.
```
Comment (aaroncarey) [inline content/releases/index.md:56]: Do we know how soon we will be able to switch Dgraph Cloud backends to "read-only" during upgrades?

File: content/releases/index.md [hunk 4]
```diff
@@ -39,3 +39,29 @@ To learn about the latest releases and other important announcements, watch the
...
+* **Dgraph Cloud rolling upgrades**: Dgraph Cloud service updates that don't
+include breaking API changes or data format changes are handled as rolling upgrades,
+with no impact on HA clusters and a few minutes of downtime for non-HA clusters.
+
+<!-- Original API deprecation wording, for reviewer reference:
```
Comment (bucanero) [inline content/releases/index.md:61]:
```suggestion
<!--
Original API deprecation wording, for reviewer reference:  
```

**Pull request closed without merging**